cmd/utils.go

```diff
@@ -95,9 +97,62 @@ func splitStr(path, sep string, n int) []string {
 	return splits
 }
 
+func buildS3ConfigFromEnv(envConfig, urlStr string) (*Config, *probe.Error) {
+	u, e := url.Parse(envConfig)
+	if e != nil {
+		return nil, probe.NewError(e).Trace(envConfig)
+	}
+	var (
+		accessKey    = u.User.Username()
+		secretKey, _ = u.User.Password()
+	)
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return nil, errInvalidArgument().Trace(u.Scheme)
+	}
+
+	hostURL, e := url.Parse(u.Scheme + "://" + u.Host)
+	if e != nil {
+		return nil, probe.NewError(e).Trace(envConfig)
+	}
+
+	s3Config, err := newS3Config(cli.Args{hostURL.String(), accessKey, secretKey})
+	if err != nil {
+		return nil, err.Trace(envConfig)
+	}
+
+	s3Config.AppName = "mc"
+	s3Config.AppVersion = Version
+	s3Config.AppComments = []string{os.Args[0], runtime.GOOS, runtime.GOARCH}
+	s3Config.Debug = globalDebug
+	s3Config.Insecure = globalInsecure
+
+	return s3Config, nil
+}
+
+func getAliasFromEnv(envs []string) string {
+	for _, k := range envs {
+		if strings.HasPrefix(k, "MC_TARGET_") {
+			k = strings.TrimPrefix(k, "MC_TARGET_")
+			return k
+		}
+	}
+	return ""
+}
+
 // buildS3Config fetches config related to the specified alias
 // to create a new config structure
 func buildS3Config(alias, urlStr string) (*Config, *probe.Error) {
+	// Check if we can get the values from ENV, if not use from `config.json`.
+	// NOTE: getAliasFromEnv() we will use alias from env if available, we will
+	// override ENV values with config entries as well.
+	if aliasEnv := getAliasFromEnv(os.Environ()); aliasEnv != "" {
+		alias = aliasEnv
```

Comment:
alias will hold `myminio=http://minio:minio123@localhost:9000`, you need further processing here.

Besides, one problem with this approach is that we can't support more than one alias, and that's the reason we have this env format MC_TARGET_alias.

I can approve this and I'll send a separate PR to address alias issue.

Reply:
oh yes you are right @vadmeste i need to filter it out.  Yes that is a limitation right now, let me see if it can be addressed properly.